src/Symfony/Component/Console/Command/InvokableCommand.php

```diff
@@ -0,0 +1,106 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Attribute\Argument;
+use Symfony\Component\Console\Attribute\Option;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Represents an invokable command.
+ *
+ * @author Yonel Ceruto <open@yceruto.dev>
+ *
+ * @internal
+ */
+class InvokableCommand
+{
+    private readonly \ReflectionFunction $reflection;
+
+    public function __construct(
+        private readonly Command $command,
+        private readonly \Closure $code,
+    ) {
+        $this->reflection = new \ReflectionFunction($code);
+    }
+
+    /**
+     * Invokes a callable with parameters generated from the input interface.
+     */
+    public function __invoke(InputInterface $input, OutputInterface $output): int
+    {
+        $statusCode = ($this->code)(...$this->getParameters($input, $output));
+
+        if (null !== $statusCode && !\is_int($statusCode)) {
+            throw new LogicException(\sprintf('The command "%s" must return either void or an integer value in the "%s" method, but "%s" was returned.', $this->command->getName(), $this->reflection->getName(), get_debug_type($statusCode)));
+        }
+
+        return $statusCode ?? 0;
+    }
+
+    /**
+     * Configures the input definition from an invokable-defined function.
+     *
+     * Processes the parameters of the reflection function to extract and
+     * add arguments or options to the provided input definition.
+     */
+    public function configure(InputDefinition $definition): void
+    {
+        foreach ($this->reflection->getParameters() as $parameter) {
+            if ($argument = Argument::tryFrom($parameter)) {
+                $definition->addArgument($argument->toInputArgument());
+            } elseif ($option = Option::tryFrom($parameter)) {
+                $definition->addOption($option->toInputOption());
+            }
+        }
+    }
+
+    private function getParameters(InputInterface $input, OutputInterface $output): array
+    {
+        $parameters = [];
+        foreach ($this->reflection->getParameters() as $parameter) {
+            if ($argument = Argument::tryFrom($parameter)) {
+                $parameters[] = $argument->resolveValue($input);
+
+                continue;
+            }
+
+            if ($option = Option::tryFrom($parameter)) {
+                $parameters[] = $option->resolveValue($input);
+
+                continue;
+            }
+
+            $type = $parameter->getType();
+
+            if (!$type instanceof \ReflectionNamedType) {
+                continue;
```

Comment:
There will be a problem if you mix parameters with a recognized type and parameters without a type.

```suggestion
                throw new RuntimeException(\sprintf('Missing type for parameter "$%s".', $parameter->getName())),
```

Reply:
We should already throw an exception when using the new invokable commands, and trigger the deprecation only for the `setCode` calls. This requires a parameter when instantiating `InvokableCommand` to distinct them.

Reply:
An invokable command can also be configured using `setCode`, but I still find your idea valid. It's been updated!